src/main/resources/com/chikli/hudson/plugin/naginator/NaginatorPublisher/config.jelly

```diff
@@ -17,13 +17,17 @@
     </f:entry>
 
     <f:advanced>
-        <f:entry field="checkRegexp"
+        <f:optionalBlock field="checkRegexp" inline="true"
                  title="${%Only rerun build if regular expression is found in output}">
-            <f:checkbox />
-        </f:entry>
-        <f:entry title="${%Regular expression to search for}" field="regexpForRerun">
-            <f:textbox />
-        </f:entry>
+            <f:entry title="${%Regular expression to search for}" field="regexpForRerun">
+                <f:textbox />
+            </f:entry>
+            <j:if test="${descriptor.matrixProject}">
+                <f:entry title="${%Test regular expression for the matrix parent}" field="regexpForMatrixParent">
+                    <f:checkbox />
+                </f:entry>
+            </j:if>
+        </f:optionalBlock>
```

Comment:
Changes:
* "Regular expression to search for" is displayed only when you checked "Only rerun build if regular expression is found in output".
* Added "Test regular expression for matrix parent", which is displayed only for matrix projects.